query/src/servers/http/v1/stage.rs

```diff
@@ -49,6 +48,8 @@ pub async fn upload_to_stage(
 
     let user_mgr = context.get_user_manager();
 
+    // TODO(xuanwo): logic here seems buggy, we need to fix it.
+    // It's incorrect to get operator from context if we are uploading an external stage.
```

Comment:
Add new function as `get_stage_operator`?

Reply:
These logics are not covered by stateful tests. So I think they are wrong for a long time.

I will fix them in another along with stateful tests.